### ISSUE TITLE
Add less restrictive methods for Array1

### ISSUE THREAD

**Issue by Robbepop** [bluss/rust-ndarray#348]
`Array1` of `ndarray` currently suffers from `ndarray`'s generic interfaces to its provided data structures.
One example are the `as_slice` and `as_slice_mut` methods that return an `Option` of a slice instead of directly returning a slice to the underlying data because in higher dimensional array this could be an invalid operation.
However, for `Array1` it never is and so users of this library could profit from a specialized implementation of some methods on `Array1` that make use of its special invariants.

**Comment by bluss** [comment]
Yes. I'm sure we talked about this before.

Your concern about the usability of the slicing methods should be taken at face value, it **is** annoying and the combination of owned storage and 1-D should mean we are contiguous. *At least* for uniquely owned (Array).

The first step would be, unfortunately, to ensure this actually holds. So we make the goal that uniquely owned 1D arrays are contiguous.

It does not hold at the moment, the way to create non contiguous `Array1` are using either [`from_shape_vec`](https://docs.rs/ndarray/0.10.5/ndarray/struct.ArrayBase.html#method.from_shape_vec) and / or [`islice`](https://docs.rs/ndarray/0.10.5/ndarray/struct.ArrayBase.html#method.islice).

It's simplest to explain with islice: if one `islice(s![..; 2])`, then the array is no longer contiguous. I can't guarantee we can actually get to the goal here.

**Comment by Robbepop** [comment]
Thank you very much for your comment!
It truely is unfortunate that `Array1` invariants aren't as I thought.
I also do not have a clear memory that we have already talked about this in detail - sorry.
But it isn't such an important feature - just another enhancement. ;)

**Comment by robsmith11** [comment]
I agree that a simpler interface for contiguous 1-dimensional arrays would be nice..
Is there a better library that focuses on linear algebra with 1 and 2-dimensional dense arrays?

**Comment by bluss** [comment]
@robsmith11 Check out rulinalg and nalgebra
